elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -694,6 +696,16 @@ private SearchResult search(boolean enableScrolling) {
 				}
 			}
 			Search search = builder.build();
+
+			if ( QUERY_LOG.isDebugEnabled() ) {
+				try ( ServiceReference<GsonService> gsonService =
+						extendedIntegrator.getServiceManager().requestReference( GsonService.class ) ) {
```

Comment:
@Sanne I just had a look at the service manager code, and I guess the efficiency issue you hinted at was with those synchronized start/stop methods in `org.hibernate.search.engine.service.impl.StandardServiceManager.ServiceWrapper`.

Couldn't we use plain `java.util.concurrent.locks.Lock`s under the hood instead of our own counters here? That way, we'd have a read lock (that would ensure nobody is going to start/stop a service while it's being used) and a write lock (that would ensure we're free to start/stop a service without blocking anyone else). I was thinking of something similar to what you did in `org.hibernate.search.backend.impl.lucene.LazyExecutorHolder`.